addons/website/static/src/client_actions/website_preview/website_preview.js

```diff
@@ -53,7 +54,16 @@ export class WebsitePreview extends Component {
             this.backendWebsiteId = unslugHtmlDataObject(backendWebsiteRepr).id;
 
             const encodedPath = encodeURIComponent(this.path);
-            if (this.websiteDomain && this.websiteDomain !== window.location.origin) {
+            if (this.websiteDomain && !wUtils.areSameDomain(this.websiteDomain, window.location.origin)) {
```

Comment:
> 3. The code in charge of loading the iframe URL (website_preview.js)
     will detect that `http` URL as originating from a different domain
     than the one of the current tab URL (parent window).
  > 4. In this case, that code will reject that URL and instead load it in
     the parent window URL -> It will basically redirect your whole
     browser tab to that URL.
     This is done as if you are loading the URL of a different website,
     we want you to be completely redirect to this other website domain.
     Also, when loading third party URL (facebook, youtube, whatever),

I don't think this part of your explanation is right (or at least it is confusing) (the last sentence): the code you are fixing is not in charge of redirecting third party URL.

Reply:
Indeed :+1: (even if same principle, this is done elsewhere through isTOpParentUrl)
Removed